backend/src/models/teams.rs

```diff
@@ -46,7 +54,247 @@ impl teams::Model {
         .await?;
 
         let mut active_user: users::ActiveModel = user.into();
-        active_user.team_name = Set(Some(team_name));
+        active_user.team_name = Set(Some(team_name.clone()));
+        active_user.leads = Set(Some(team_name));
+        active_user.update(&transaction).await?;
+
+        transaction.commit().await?;
+
+        Ok(())
+    }
+    pub async fn invite_user(
+        database: &DatabaseConnection,
+        add_user_json: AddUserModel,
+        claim_data: Claims,
+    ) -> Result<(), Error> {
+        let user = users::Entity::find_by_id(&claim_data.email)
+            .one(database)
+            .await?
+            .ok_or(Error::Unauthorized)?;
+
+        if user.team_name.is_none() {
+            return Err(Error::Team(UserDoesntBelongToAnyTeam));
+        }
+
+        let team = teams::Entity::find()
+            .filter(teams::Column::Name.eq(user.team_name.unwrap()))
+            .one(database)
+            .await?
+            .ok_or(Error::Team(TeamNotFound))?;
+
+        if user.username != team.leader_name {
+            return Err(Error::Unauthorized);
+        }
+
+        let invited_user = users::Entity::find()
+            .filter(users::Column::Username.eq(&add_user_json.username))
+            .one(database)
+            .await?
+            .ok_or(Error::Auth(UserNotFound))?;
```

Comment:
The point of those new errors was to use auth errors only in auth endpoint. If you want to use usernotfound in multiple places, move it to common one